### PR TITLE
feat: build arm64 Docker image for amd64/arm64

### PR DESCRIPTION
**Pull request by kuisathaverat**
build arm64 Docker image for amd64/arm64

depends on https://github.com/elastic/golang-crossbuild/pull/197
related to https://github.com/elastic/golang-crossbuild/issues/153

### CODE REVIEW

**Review by v1v — COMMENTED** [review]
IIUC, the same Dockerfile.tmp is used for building on ARM and AMD, and `TARGETOS` and `TARGETARCH` are the variables that drive what's the final docker image.

**Comment by kuisathaverat** [comment]
>IIUC, the same Dockerfile.tmp is used for building on ARM and AMD, and TARGETOS and TARGETARCH are the variables that drive what's the final docker image.

they are not when you build with the build you will launch two processes with different platforms, in this Docker file you will choose two different Dockerfile parts and those variables will be different on both https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope

File: go/arm/Dockerfile.tmpl [hunk 1]
```diff
@@ -46,6 +50,62 @@ RUN apt install -y \
...
 {{- end }}
 
+ARG REPOSITORY
+ARG VERSION
+ARG TAG_EXTENSION=''
+FROM --platform=linux/arm64 ${REPOSITORY}/golang-crossbuild:${VERSION}-base${TAG_EXTENSION} as stage-arm64
+
+ARG TARGETOS
+ARG TARGETARCH
+RUN echo "I'm building for $TARGETOS/$TARGETARCH"
```
Comment (kuisathaverat) [inline go/arm/Dockerfile.tmpl:60]: TODO remove on the final version

File: go/arm/Dockerfile.tmpl [hunk 2]
```diff
@@ -1,7 +1,11 @@
 ARG REPOSITORY
 ARG VERSION
 ARG TAG_EXTENSION=''
-FROM ${REPOSITORY}/golang-crossbuild:${VERSION}-base${TAG_EXTENSION}
+FROM --platform=linux/amd64 ${REPOSITORY}/golang-crossbuild:${VERSION}-base${TAG_EXTENSION} as stage-amd64
+
+ARG TARGETOS
+ARG TARGETARCH
+RUN echo "I'm building for $TARGETOS/$TARGETARCH"
```
Comment (kuisathaverat) [inline go/arm/Dockerfile.tmpl:8]: TODO remove on the final version

File: go/arm/Makefile [hunk 3]
```diff
@@ -1 +1,8 @@
 include ../Makefile.common
+
+ifeq ($(DOCKER_MULTIARCH),1)
+DOCKER_CMD := $(SELF_DIR)/../.ci/scripts/buildx.sh
+
+push:
+	@echo "Already done."
```
Comment (kuisathaverat) [inline go/arm/Makefile:7]:
```suggestion
	@echo "Already done by buildx (.ci/scripts/buildx.sh)."
```

File: go/arm/Makefile [hunk 4]
```diff
@@ -1 +1,8 @@
 include ../Makefile.common
+
+ifeq ($(DOCKER_MULTIARCH),1)
+DOCKER_CMD := $(SELF_DIR)/../.ci/scripts/buildx.sh
+
+push:
+	@echo "Already done by buildx (.ci/scripts/buildx.sh)."
```
Comment (kuisathaverat) [inline go/arm/Makefile:7]:
```suggestion
```suggestion
	@echo "Already done by buildx (.ci/scripts/buildx.sh)."
atomic-push:
	@echo "Already done by buildx (.ci/scripts/buildx.sh)."
```
```

**Pull request merged**